examples/provider/provider.tf

```diff
@@ -1,8 +1,20 @@
+# Provider configuration for TMC SaaS
 provider "tanzu-mission-control" {
   endpoint            = var.endpoint            # optionally use TMC_ENDPOINT env var
   vmw_cloud_api_token = var.vmw_cloud_api_token # optionally use VMW_CLOUD_API_TOKEN env var
 
   # if you are using dev or different csp endpoint, change the default value below
   # for production environments the vmw_cloud_endpoint is console.cloud.vmware.com
   # vmw_cloud_endpoint = "console.cloud.vmware.com" or optionally use VMW_CLOUD_ENDPOINT env var
+}
+
+# Provider configuration for TMC Self-Managed
```

Comment:
Kindly add the examples in the `docs/index.md` file by adding it to the templates and generating the documentation.

Reply:
Documentation is a different task and not in the scope of this MR

Reply:
Can you please link the JIRA or any open MR for the same here

Reply:
This is github and considering it to be opensource I do not think linking the Jira task ID which is internal is best practice. 
and I do not have any open MRs. 
PS: This MR is getting merged into feature branch and not master.